File: phpopencloud/firstapp-getting-started_10.php
#! /usr/bin/env php
<?php

require_once __DIR__ . '/vendor/autoload.php';//'php-opencloud/openstack/src/OpenStack.php';

use Symfony\Component\Yaml\Yaml;

try {  $configs = Yaml::parse(file_get_contents('clouds.yaml'));}
catch (ParseException $e) { printf("Unable to parse the YAML string: %s", $e->getMessage()); }

$openstack = new OpenStack\OpenStack([
    'authUrl' => $configs[ 'clouds' ][ 'internap' ][ 'authUrlV3'],
    'region'  => $configs[ 'clouds' ][ 'internap' ][ 'region_name' ],
    'user'    => [
        'name'      => $configs[ 'clouds' ][ 'internap' ][ 'auth' ][ 'username' ],
        'password'  => $configs[ 'clouds' ][ 'internap' ][ 'auth' ][ 'password' ],
        'domain'    => [
            'name'    => $configs[ 'clouds' ][ 'internap' ][ 'auth' ][ 'domain_name' ]
        ]
    ]
]);

$security_group_name        = 'all-in-one';
$security_group_description = 'network access for all-in-one application.';

# step-9

$neutron = $openstack->networkingV2ExtSecGroups();

$security_group_exists = False;
echo 'Checking for existing security group named ' . $security_group_name . "...\n";
foreach ($neutron->listSecurityGroups() as $secGroup) {
  if ($secGroup->name == $security_group_name)
      $security_group_exists = True;
}

if ($security_group_exists) {
    echo " \tsecurity group " . $security_group_name . " already exists, skipping creation.\n";
} else {
    echo " \tadding security group...";
    $secGroup = $neutron->createSecurityGroup([
        'name'        => $security_group_name,
        'description' => $security_group_description,
    ]);
}

echo( "Listing security groups...\n" );
foreach ($neutron->listSecurityGroups() as $secGroup) {
    echo ' - found security group with ID ' . $secGroup->id . ' and name ' .  $secGroup->name . "\n";
}

echo( "Done! Congrats\n" );

File: shade/firstapp-getting-started_10.py
from shade import *


simple_logging(debug=True)
conn = openstack_cloud(cloud='internapNYJ')

#step-10
print('Checking for existing security groups...')
sec_group_name = 'all-in-one'
if conn.search_security_groups(sec_group_name):
    print('Security group already exists. Skipping creation.')
else:
    print('Creating security group.')
    conn.create_security_group(sec_group_name, 'network access for all-in-one application.')
    conn.create_security_group_rule(sec_group_name, 80, 80, 'TCP')
    conn.create_security_group_rule(sec_group_name, 22, 22, 'TCP')

secgroup = conn.search_security_groups(sec_group_name)
print( secgroup )

print( 'Done! Congrats!')
